### PR TITLE
Sync Progress as a Percentage In Initial Sync

### LINKED ISSUE #2150: Implement Sync Progress as a Percentage during initial sync
**Describe the feature request.**
At the moment, initial sync progress is shown by an indeterminate spinning circular progress bar. We would like to incorporate the progress as a percentage for a better UX.

**Additional context**
FHIR SDK now supports this by providing the totals via the **In Progress** sync status updates

**Acceptance criteria**
During initial sync the progress dialog should show the progress as a percentage.

**Area path**
1. Perform a fresh install
2. Log in the app to trigger the initial sync

**Implementation plan (For Engineers)**
- [ ] Publish the dependent FHIR SDK library artifacts
- [ ] Enhance the current **LoaderView** dialog implementation to render progress as a percentage

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Closes #2150 
- Additionally fixes issue where Sync Dialog **does NOT** dismiss immediately after completing sync and render register contents _(Infinity sync dialog)_

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ndegwamartin** [comment]
Screenshot_20230313_154333

**Review by f-odhiambo — DISMISSED** [review]
I can confirm that this is working as described running this on build variant

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt [hunk 1]
```diff
@@ -184,6 +184,12 @@ class RegisterFragment : Fragment(), OnSyncListener, Observer<QuestionnaireSubmi
             SnackBarMessageConfig(message = getString(R.string.syncing))
           )
         }
+      is SyncJobStatus.InProgress ->
+        lifecycleScope.launch {
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt:188]: is it possible to write a test that gets at this? Lifecycle stuff seems hard to test

Comment (ndegwamartin) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterFragment.kt]: A little refactoring has got us the desired result :)

**Review by pld — APPROVED** [review]

**Pull request merged**